synapseclient/core/multithread_download/download_threads.py

```diff
@@ -0,0 +1,396 @@
+import time
+
+try:
+    import threading as _threading
+except ImportError:
+    import dummy_threading as _threading
+import datetime
+import os
+import queue
+
+from typing import Generator, Sequence, NamedTuple, Tuple, Iterable
+from urllib.parse import urlparse, parse_qs
+from urllib3.util.retry import Retry
+from synapseclient.core.utils import printTransferProgress
+from synapseclient.core.exceptions import SynapseError
+from requests import Session, Response
+from requests.adapters import HTTPAdapter
+from http import HTTPStatus
+
+# constants
+MAX_QUEUE_SIZE: int = 20
+MAX_RETRIES: int = 20
+MiB: int = 2 ** 20
+SYNAPSE_DEFAULT_DOWNLOAD_PART_SIZE: int = 8 * MiB
+MAX_CHUNK_WRITE_SIZE = 2 * MiB
+ISO_AWS_STR_FORMAT: str = '%Y%m%dT%H%M%SZ'
+CONNECT_FACTOR: int = 3
+BACK_OFF_FACTOR: float = 0.5
+
+
+class DownloadRequest(NamedTuple):
+    """
+    A request to download a file from Synapse
+
+    ...
+
+    Attributes
```

Comment:
Inconsistent documentation method in the client.  This is a `numpy` style, but we use the `sphinx` style.  Although personally, I like the google/numpy style docs better.

Reply:
The core package code is not user facing and won't have sphinx docs generated for it.